### PR TITLE
ICU-21480 Change ICU 64 draft readResolve APIs to @internal

### PR DESCRIPTION
**Pull request by sven-oly**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21480
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4j/main/classes/core/src/com/ibm/icu/text/DateIntervalFormat.java [hunk 1]
```diff
@@ -377,7 +377,7 @@ private SpanField(String name) {
         /**
          * serialization method resolve instances to the constant
          * DateIntervalFormat.SpanField values
-         * @draft ICU 64
+         * @internal ICU 64
```
[the remark this answers is not in the document]
Comment (sven-oly) [reply icu4j/main/classes/core/src/com/ibm/icu/text/DateIntervalFormat.java]: Thanks for the additional info. PTAL.

File: icu4j/main/classes/core/src/com/ibm/icu/text/DateIntervalFormat.java [hunk 2]
```diff
@@ -377,7 +377,8 @@ private SpanField(String name) {
         /**
          * serialization method resolve instances to the constant
          * DateIntervalFormat.SpanField values
-         * @draft ICU 64
+         * @internal
+         * @deprecated This API is ICU internal only.
          */
         @Override
```
Comment (yumaoka) [inline icu4j/main/classes/core/src/com/ibm/icu/text/DateIntervalFormat.java:383]: As I mentioned my previous comment, you need @Deprecated annotation here.

--------
        /**
         * serialization method resolve instances to the constant
         * DateIntervalFormat.SpanField values
         * @draft ICU 64
         * @internal
         * @deprecated This API is ICU internal only.
         */
        @Deprecated
        @Override
--------

See `@Deprecated` out of API comment, next to `@Override`.  It looks this is not explained in the same BRS task section - I'll update the description.

**Review by yumaoka — CHANGES_REQUESTED** [review]
We need `@Deprecated` annotation (not just JavaDoc `@deprecated` tag) as well.

File: icu4j/main/classes/core/src/com/ibm/icu/text/DateIntervalFormat.java (hunk 2 above)
Comment (sven-oly) [reply icu4j/main/classes/core/src/com/ibm/icu/text/DateIntervalFormat.java:384]: Thanks for the additional explanation. I've updated these 3 files.

**Comment by sven-oly** [comment]
> We need `@Deprecated` annotation (not just JavaDoc `@deprecated` tag) as well.

Updated. Ready for another look.

**Review by markusicu — APPROVED** [review]
lgtm tnx

**Comment by sven-oly** [comment]
> We need `@Deprecated` annotation (not just JavaDoc `@deprecated` tag) as well.

Done.

**Review by yumaoka — APPROVED** [review]
LGTM

**Pull request merged**